tests/libvmifact/architecture.go

```diff
@@ -19,12 +19,69 @@
 
 package libvmifact
 
-var architecture string
+var (
+	architecture   string
+	minMemProvider MinimalMemoryProvider
```

Comment:
since we don't have any other provider besides memory, I would just call it ArchProvider and postpone further specialization to a future point in time where it's needed.

Reply:
In golang, we're trying to keep interfaces as small as possible, preferring one method. Here we have already two methods. My concern is if we'll have a generic name, it will start getting too many methods.

So if we'll want to add new behavior from different aspect, I prefer to have new interfaces, and let the providers to implement multiple interfaces.

Reply:
what I did eventually is to add the archProvider interface, to include the memoryProvider interface. Then we could add more interfaces in the future:
```go
type archProvider interface {
	minimalMemoryProvider
}

// minimalMemoryProvider is an interface to return the minimal memory needed for a VMI, according to the architecture
type minimalMemoryProvider interface {
	getMinimalMemory() string
	getQemuMinimalMemory() string
}
```